vector/src/main/java/im/vector/app/features/roomprofile/notifications/RoomNotificationSettingsFragment.kt

```diff
@@ -0,0 +1,90 @@
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package im.vector.app.features.roomprofile.notifications
+
+import android.os.Bundle
+import android.view.LayoutInflater
+import android.view.View
+import android.view.ViewGroup
+import androidx.core.view.isVisible
+import com.airbnb.mvrx.fragmentViewModel
+import com.airbnb.mvrx.withState
+import im.vector.app.R
+import im.vector.app.core.extensions.configureWith
+import im.vector.app.core.platform.VectorBaseFragment
+import im.vector.app.databinding.FragmentRoomSettingGenericBinding
+import im.vector.app.features.home.AvatarRenderer
+import org.matrix.android.sdk.api.session.room.notification.RoomNotificationState
+import org.matrix.android.sdk.api.util.toMatrixItem
+import javax.inject.Inject
+
+class RoomNotificationSettingsFragment @Inject constructor(
+        val viewModelFactory: RoomNotificationSettingsViewModel.Factory,
+        private val roomNotificationSettingsController: RoomNotificationSettingsController,
+        private val avatarRenderer: AvatarRenderer
+) : VectorBaseFragment<FragmentRoomSettingGenericBinding>(),
+        RoomNotificationSettingsController.Callback {
+
+    private val viewModel: RoomNotificationSettingsViewModel by fragmentViewModel()
+
+    override fun getBinding(inflater: LayoutInflater, container: ViewGroup?): FragmentRoomSettingGenericBinding {
+        return FragmentRoomSettingGenericBinding.inflate(inflater, container, false)
+    }
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        super.onViewCreated(view, savedInstanceState)
+        setupToolbar(views.roomSettingsToolbar)
+        roomNotificationSettingsController.callback = this
+        views.roomSettingsRecyclerView.configureWith(roomNotificationSettingsController, hasFixedSize = true)
+        setupWaitingView()
+        observeViewEvents()
+    }
+    private fun setupWaitingView() {
+        views.waitingView.waitingStatusText.setText(R.string.please_wait)
+        views.waitingView.waitingStatusText.isVisible = true
+    }
+
+    private fun observeViewEvents() {
+        viewModel.observeViewEvents {
+            when (it) {
+                is RoomNotificationSettingsViewEvents.Failure -> displayErrorDialog(it.throwable)
+            }
+        }
+    }
+
+    override fun invalidate() = withState(viewModel) { viewState ->
+        roomNotificationSettingsController.setData(viewState)
+        views.waitingView.root.isVisible = viewState.isLoading
+        renderRoomSummary(viewState)
+    }
+
+    override fun didSelectRoomNotificationState(roomNotificationState: RoomNotificationState) {
+        viewModel.handle(RoomNotificationSettingsAction.SelectNotificationState(roomNotificationState))
+    }
+
+    override fun didSelectAccountSettingsLink() {
+        navigator.openSettings(requireContext())
```

Comment:
Pass a second parameter `EXTRA_DIRECT_ACCESS_NOTIFICATIONS` to directly open the sub screen?

Reply:
Yea wasn't sure which based on the text in the message. I'll run it by design.

Reply:
Yes, it's the account setting regarding the notifications, so I guess it's OK to ease the user life by opening directly the sub setting screen